incubator/virtualcluster/experiment/pkg/scheduler/cache/snapshot.go

```diff
@@ -135,3 +135,58 @@ func (c *schedulerCache) SnapshotForNamespaceSched(nsToRemove ...*Namespace) (*N
 	}
 	return s, nil
 }
+
+type PodSchedSnapshot struct {
+	clusterUsageMap map[string]*ClusterUsage
+}
+
+func NewPodSchedSnapshot() *PodSchedSnapshot {
+	return &PodSchedSnapshot{
+		clusterUsageMap: make(map[string]*ClusterUsage),
+	}
+}
+
+func (s *PodSchedSnapshot) GetClusterUsageMap() map[string]*ClusterUsage {
+	return s.clusterUsageMap
+}
+
+func (s *PodSchedSnapshot) AddUsage(cluster string, usage v1.ResourceList) error {
+	cur, exists := s.clusterUsageMap[cluster]
+	if !exists {
+		return fmt.Errorf("slices are added to nonexistence cluster")
+	}
+	for k, v := range usage {
+		val := cur.alloc[k].DeepCopy()
+		val.Add(v)
+		cur.alloc[k] = val
+	}
+	return nil
+}
+
+func (c *schedulerCache) SnapshotForPodSched(pod *Pod) (*PodSchedSnapshot, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	s := NewPodSchedSnapshot()
+	nsKey := pod.GetNamespaceKey()
+
+	ns := c.namespaces[nsKey]
+	if ns == nil {
+		return nil, fmt.Errorf("ns %s not found", nsKey)
+	}
+
+	for _, place := range ns.schedule {
+		capability := v1.ResourceList{}
+
+		for k, v := range ns.quotaSlice {
+			val := v.DeepCopy()
+			val.Set(v.Value() * int64(place.num))
+			capability[k] = val
+		}
+		s.clusterUsageMap[place.cluster] = &ClusterUsage{
+			capacity: capability,
```

Comment:
Where do track the allocated capacity for each cluster? I suppose you need to traverse the Pod map in each cluster.